src/main/kotlin/com/autonomousapps/extension/Behavior.kt

```diff
@@ -2,7 +2,11 @@ package com.autonomousapps.extension
 
 import java.io.Serializable
 
-sealed class Behavior(val filter: Set<String> = setOf()) : Serializable, Comparable<Behavior> {
+sealed class Behavior(
+  val filter: Set<String> = setOf(),
+  val sourceSetName: String = Issue.ALL_SOURCE_SETS
```

Comment:
The core of the new implementation is embedding source set information in the model here.